packages/compiler/core/parser.ts

```diff
@@ -932,14 +944,14 @@ export function parse(code: string | SourceFile, options: ParseOptions = {}): Ca
     // identifier. We want to parse `@ model Foo` as invalid decorator
     // `@<missing identifier>` applied to `model Foo`, and not as `@model`
     // applied to invalid statement `Foo`.
-    const target = parseIdentifierOrMemberExpression(undefined, false);
+    const target = parseIdentifierOrMemberExpression(undefined, true);
```

Comment:
This looks like the source of the decorator completion failure. Exactly what the comment says applies 😁